### ISSUE TITLE
We do not inform user whenever remotes are started to get ignored upon get

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#3564]
TL;DR:  there might be a bug with parsing/reporting multiple lines in stderr while operating  in --json mode.    git-annex seems to spit out the message that "not usable by git-annex; setting annex-ignore" to stderr but I do not observe it in our logs

I pointed datasets.datalad.org to localhost in /etc/hosts, so no website is reachable by that name
```shell
hopa:/tmp
$> wget datasets.datalad.org
--2019-07-26 13:20:46--  http://datasets.datalad.org/
Resolving datasets.datalad.org (datasets.datalad.org)... 127.0.1.1
Connecting to datasets.datalad.org (datasets.datalad.org)|127.0.1.1|:80... failed: Connection refused.
```
then I am installing ReproNim/containers from github, during which git-annex adds datasets.datalad.org remote but doesn't contact it to obtain UUID.  Only later when I try to `get` the file, it tries to enable it but fails but we log (INFO like we do for initial clone from github or even WARNING in this particular case) that special remote was not enabled:

```shell
$> datalad install https://github.com/ReproNim/containers
[INFO   ] Cloning https://github.com/ReproNim/containers [1 other candidates] into '/tmp/containers' 
[INFO   ]   Remote origin not usable by git-annex; setting annex-ignore                                                        
[INFO   ] access to dataset sibling "datasets.datalad.org" not auto-enabled, enable with:
| 		datalad siblings -d "/tmp/containers" enable -s datasets.datalad.org 
install(ok): /tmp/containers (dataset)

$> git -C containers config remote.datasets.datalad.org.annex-ignore       

$> datalad get containers/images/bids/bids-validator--1.2.3.sing
[WARNING] Running get resulted in stderr output: download failed: Unauthorized
git-annex: get: 1 failed
 
[ERROR  ] from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web;  	23456fab-05d4-441e-8952-b8c6d90ad785 -- yoh@smaug:~/proj/repronim/containers;  	3bcd23f7-1adf-4dce-9f2a-efdaab4151d8 -- yoh@hopa:~/proj/repronim/containers;  	71c620b5-997f-4849-bb30-c42dbb48a51e -- datasets.datalad.org; ; (Note that these git remotes have annex-ignore set: datasets.datalad.org origin) [get(/tmp/containers/images/bids/bids-validator--1.2.3.sing)] 
get(error): /tmp/containers/images/bids/bids-validator--1.2.3.sing (file) [from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web;  	23456fab-05d4-441e-8952-b8c6d90ad785 -- yoh@smaug:~/proj/repronim/containers;  	3bcd23f7-1adf-4dce-9f2a-efdaab4151d8 -- yoh@hopa:~/proj/repronim/containers;  	71c620b5-997f-4849-bb30-c42dbb48a51e -- datasets.datalad.org; ; (Note that these git remotes have annex-ignore set: datasets.datalad.org origin)]

$> git -C containers config remote.datasets.datalad.org.annex-ignore
true
```

So we do receive/report that Unauthorized stderr but somehow there is nothing about datasets.datalad.org obtaining annex-ignore.

if I redo (clean install) git-annex call directly -- message seems to be there before Unauthorized:

```
/tmp > git -C containers annex get --json --json-progress -J4 images/bids/bids-validator--1.2.3.sing >/dev/null 
  Remote datasets.datalad.org not usable by git-annex; setting annex-ignore
download failed: Unauthorized
git-annex: get: 1 failed
```
so either we filter it out or just somehow drop it?

```shell
~datalad/datalad > git describe
0.11.5-56-g85a2cf994

~datalad/datalad > git annex version
git-annex version: 7.20190708+git36-g32d526164-1~ndall+1
```

**Comment by kyleam** [comment]
If I try your recipe now (on master, 2450ab986), `datalad get` output includes "Remote datasets.datalad.org not usable by git-annex; setting annex-ignore".

```
% datalad get containers/images/bids/bids-validator--1.2.3.sing
[WARNING] Running get resulted in stderr output:   Remote datasets.datalad.org not usable by git-annex; setting annex-ignore
download failed: Unauthorized
git-annex: get: 1 failed
 
[ERROR  ] from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web;  	23456fab-05d4-441e-8952-b8c6d90ad785 -- yoh@smaug:~/proj/repronim/containers;  	3bcd23f7-1adf-4dce-9f2a-efdaab4151d8 -- yoh@hopa:~/proj/repronim/containers;  	71c620b5-997f-4849-bb30-c42dbb48a51e -- datasets.datalad.org; ; (Note that these git remotes have annex-ignore set: datasets.datalad.org origin) [get(/tmp/dl-9rxwRig/containers/images/bids/bids-validator--1.2.3.sing)] 
get(error): /tmp/dl-9rxwRig/containers/images/bids/bids-validator--1.2.3.sing (file) [from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web;  	23456fab-05d4-441e-8952-b8c6d90ad785 -- yoh@smaug:~/proj/repronim/containers;  	3bcd23f7-1adf-4dce-9f2a-efdaab4151d8 -- yoh@hopa:~/proj/repronim/containers;  	71c620b5-997f-4849-bb30-c42dbb48a51e -- datasets.datalad.org; ; (Note that these git remotes have annex-ignore set: datasets.datalad.org origin)]
```

Do you still see the issue?  Perhaps it was caused by 9d0665437 (BF: cmd: Don't leave unread lines on stream when logging online, 2019-09-09).

**Comment by yarikoptic** [comment]
Crap, probably my umask screwed things up, I will check, thanks!!

**Comment by yarikoptic** [comment]
oh, sorry -- I misintepreted the situation, Friday.  
I went back to datalad 0.11.5-56-g85a2cf994 and for me it also showed the "... not usable by git-annex ..." so it is not necessarily datalad which fixed it or may be it is a flaky issue. downgrade of git annex to 7.20190819+git2-g908476a9b-1~ndall+1 (which is later than original one I used) -- the same. so, who knows. Let's consider it dealt with and reopen when we observe it again
